src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java

```diff
@@ -172,7 +174,10 @@ public VCloudDirectorComputeServiceAdapter(VCloudDirectorApi api) {
             logger.trace("<< vApp(%s) deployment completed(%s)", vApp.getId(), vmReady);
          }
       }
-      Vm vm = Iterables.getOnlyElement(api.getVAppApi().get(vApp.getHref()).getChildren().getVms());
+      URI vAppHref = checkNotNull(vApp.getHref(), format("vApp %s must not have a null href", vApp.getId()));
```

Comment:
Could we say anything else other than "must not have a null"? For example, any idea what might have caused this?

We'll still fail with an NPE (as per https://github.com/cloudsoft/jclouds-vcloud-director/issues/11) - it will have a nicer message, but it is still a very low level message.

Is there anything else we can do, such as a retry? Or encourage Brooklyn users to do a retry of the VM provisioning by setting machineCreateAttempts to more than 1?